src/dso_api/dynamic_api/serializers.py

```diff
@@ -512,26 +558,23 @@ def serializer_factory(
         raise ValueError("flat should be False when nesting exceeds max nesting level")
 
     _validate_model(model)
-    prefix = "Flat" if flat else ""
-
-    if model.has_parent_table():
-        # Inner tables have no schema or links defined
-        fields = []
-    else:
-        fields = [
-            "_links",
-        ]
 
+    # Get model data
+    prefix = "Flat" if flat else ""
     safe_dataset_id = to_snake_case(model.get_dataset_id())
     serializer_name = f"{prefix}{safe_dataset_id.title()}{model.__name__}Serializer"
     table_schema = model.table_schema()
-    new_attrs = {
-        "table_schema": table_schema,
-        "__module__": f"dso_api.dynamic_api.serializers.{safe_dataset_id}",
-        "__doc__": table_schema.get(
-            "description", table_schema.name
-        ),  # avoid exposing our docstrings.
-    }
+
+    serializer_part = SerializerAssemblyLine(model, depth=depth)
+
+    # Generate a proper docstring for the OpenAPI spec
+    serializer_part.class_attrs["__doc__"] = table_schema.get("description", table_schema.name)
```

Comment:
Why not doing this inside `SerializerAssemblyLine`? I saw the comment about not exposing the docstring. Does doing it like this avoid exposing the docstring?

Reply:
That class is reused for more serializer construction, including the `_links` serializer and through serializer. I didn't want to render those docs for these classes as well.

Reply:
I've moved this to a constructor argument